File: viewModelTest/src/main/java/com/example/myapplication/MyAdapter.java
package com.example.myapplication;

import android.arch.lifecycle.LiveData;
import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;

/**
 * Created on 2017/11/27.
 *
 * @author 00224524
 */

public class MyAdapter extends RecyclerView.Adapter<MyAdapter.MyViewHolder> {
    private List<Person> mData;
    private Context mContext;


    protected MyAdapter(Context context, List<Person> list) {
        this.mData = list;
        this.mContext = context;
    }


    @Override
    public MyViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(mContext).inflate(R.layout.item, parent, false);
        return new MyViewHolder(view);
    }

    @Override
    public void onBindViewHolder(MyViewHolder holder, int position) {
        holder.nameTv.setText(mData.get(position).getName());
    }

    @Override
    public int getItemCount() {
        return mData.size();
    }

    class MyViewHolder extends RecyclerView.ViewHolder {

        @BindView(R.id.name_tv)
        TextView nameTv;


        MyViewHolder(View view) {
            super(view);
            ButterKnife.bind(this, view);
        }

    }
}

File: kotlinDemo/src/main/java/com/example/myapplication/MyAdapter.kt
package com.example.myapplication

import android.content.Context
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.TextView

/**
 * Created on 2018/3/28.
 * @author 00224524
 */
class MyAdapter : BaseAdapter {
    private var mData: ArrayList<String>
    private var mContext: Context

    constructor(list: ArrayList<String>, context: Context) {
        this.mData = list
        this.mContext = context
    }


    override fun getView(position: Int, convertView: View?, parent: ViewGroup?): View {
        val holder: MyViewHolder
        val v: View
        if (convertView == null) {
            v = View.inflate(mContext, R.layout.item, null)
            holder = MyViewHolder(v)
            v.tag = holder
        } else {
            v = convertView
            holder = v.tag as MyViewHolder
        }
        holder.str.text = mData[position]
        return v
    }

    override fun getItem(position: Int): Any {
        return mData[position]
    }

    override fun getItemId(position: Int): Long {
        return position.toLong()
    }

    override fun getCount(): Int {
        return mData.size
    }


    private class MyViewHolder(viewItem: View) {
        var str: TextView = viewItem.findViewById(R.id.textView) as TextView
    }
}